### PR TITLE
Update node visit order 

### PR DESCRIPTION
**Pull request by dekey**

### CODE REVIEW

**Comment by dmaicher** [comment]
@dekey can you cover this exact scenario with a functional test case? Should be possible to add something inside the `TestMenuTreeBuilder` and some test asserts?

**Comment by dekey** [comment]
@dmaicher Done.

File: tests/Functional/MenuTree/MenuTreeTraverserTest.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Tests\Functional\MenuTree;
+
+use DAMA\MenuBundle\Menu\MenuFactoryInterface;
+use DAMA\MenuBundle\Node\Node;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+class MenuTreeTraverserTest extends WebTestCase
```
Comment (dmaicher) [inline tests/Functional/MenuTree/MenuTreeTraverserTest.php]: actually I don't think we need this separate test :blush: 

You can just do the same assertions also on the DOM crawler inside `FunctionalTest`?

Comment (dekey) [reply tests/Functional/MenuTree/MenuTreeTraverserTest.php]: Ok

**Pull request merged**